bundles/org.openhab.binding.bluetooth.daikinmadoka/src/main/java/org/openhab/binding/bluetooth/daikinmadoka/handler/DaikinMadokaHandler.java

```diff
@@ -0,0 +1,761 @@
+import org.openhab.binding.bluetooth.BluetoothDevice.ConnectionState;
+import org.openhab.binding.bluetooth.ConnectedBluetoothHandler;
+import org.openhab.binding.bluetooth.daikinmadoka.DaikinMadokaBindingConstants;
+import org.openhab.binding.bluetooth.daikinmadoka.internal.BRC1HUartProcessor;
+import org.openhab.binding.bluetooth.daikinmadoka.internal.DaikinMadokaConfiguration;
+import org.openhab.binding.bluetooth.daikinmadoka.internal.model.MadokaMessage;
+import org.openhab.binding.bluetooth.daikinmadoka.internal.model.MadokaParsingException;
+import org.openhab.binding.bluetooth.daikinmadoka.internal.model.MadokaProperties.FanSpeed;
+import org.openhab.binding.bluetooth.daikinmadoka.internal.model.MadokaProperties.OperationMode;
+import org.openhab.binding.bluetooth.daikinmadoka.internal.model.MadokaSettings;
+import org.openhab.binding.bluetooth.daikinmadoka.internal.model.commands.BRC1HCommand;
+import org.openhab.binding.bluetooth.daikinmadoka.internal.model.commands.GetFanspeedCommand;
+import org.openhab.binding.bluetooth.daikinmadoka.internal.model.commands.GetIndoorOutoorTemperatures;
+import org.openhab.binding.bluetooth.daikinmadoka.internal.model.commands.GetOperationmodeCommand;
+import org.openhab.binding.bluetooth.daikinmadoka.internal.model.commands.GetPowerstateCommand;
+import org.openhab.binding.bluetooth.daikinmadoka.internal.model.commands.GetSetpointCommand;
+import org.openhab.binding.bluetooth.daikinmadoka.internal.model.commands.GetVersionCommand;
+import org.openhab.binding.bluetooth.daikinmadoka.internal.model.commands.ResponseListener;
+import org.openhab.binding.bluetooth.daikinmadoka.internal.model.commands.SetFanspeedCommand;
+import org.openhab.binding.bluetooth.daikinmadoka.internal.model.commands.SetOperationmodeCommand;
+import org.openhab.binding.bluetooth.daikinmadoka.internal.model.commands.SetPowerstateCommand;
+import org.openhab.binding.bluetooth.daikinmadoka.internal.model.commands.SetSetpointCommand;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link DaikinMadokaHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Benjamin Lafois - Initial contribution
+ */
+@NonNullByDefault
+public class DaikinMadokaHandler extends ConnectedBluetoothHandler implements ResponseListener {
+
+    private final Logger logger = LoggerFactory.getLogger(DaikinMadokaHandler.class);
+
+    private @Nullable DaikinMadokaConfiguration config;
+
+    private @Nullable ExecutorService commandExecutor;
+
+    private @Nullable ScheduledFuture<?> refreshJob;
+
+    // UART Processor is in charge of reassembling chunks
+    private BRC1HUartProcessor uartProcessor = new BRC1HUartProcessor(this);
+
+    private volatile @Nullable BRC1HCommand currentCommand = null;
+
+    private MadokaSettings madokaSettings = new MadokaSettings();
+
+    public DaikinMadokaHandler(Thing thing) {
+        super(thing);
+    }
+
+    @SuppressWarnings("null")
```

Comment:
Please try not to suppress any warnings. What is preventing you from removing this?